### PR TITLE
DEVELOPER-4542: Fix e2e cross browser tests

### PR DESCRIPTION
**Pull request by RubyGemTSL**
[DEVELOPER-4542](https://issues.jboss.org/browse/DEVELOPER-4542)

### CODE REVIEW

File: _tests/e2e/config/chimp-local-docker.js [hunk 1]
```diff
@@ -1,34 +1,22 @@
...
-    } catch (e) {
-        fs.mkdirSync(directory);
-    }
-}
-
 if (typeof process.env.RHD_BASE_URL !== 'undefined') {
     baseUrl = process.env.RHD_BASE_URL
 } else {
-    process.env.RHD_BASE_URL = 'https://developers.stage.redhat.com';
+    process.env.RHD_BASE_URL = 'https://developers-pr.stage.redhat.com/pr/2143/export';
```
Comment (LightGuard) [inline _tests/e2e/config/chimp-local-docker.js]: This doesn't look right. Looks like a hardcoded path slipped through.

File: _tests/e2e/features/step_definitions/hooks.js [hunk 2]
```diff
@@ -15,26 +15,26 @@ const hooks = function () {
...
+        //     let dirSize = [];
+        //     fs.readdirSync(pathToChromeDownloads).forEach(file => {
+        //         dirSize.push(file)
+        //     });
+        //     if (dirSize.length > 0) {
+        //         fs.emptyDir(pathToChromeDownloads)
+        //     }
+        // }
+        driver.execute('window.localStorage.clear();');
+    })
```
Comment (LightGuard) [inline _tests/e2e/features/step_definitions/hooks.js]: Any reason to keep this in instead of deleting it?

File: _tests/e2e/Dockerfile [hunk 3]
```diff
@@ -1,7 +1,7 @@
 FROM test-base:2.3.0
 
 WORKDIR /tmp
-RUN yum install -y bzip2 fontconfig freetype freetype-devel fontconfig-devel libstdc++ && \
+RUN yum install -y bzip2 fontconfig freetype freetype-devel fontconfig-devel libstdc++ unzip && \
```
Comment (LightGuard) [inline _tests/e2e/Dockerfile]: Ideally, this would include a `&& yum clean all`

File: _tests/e2e/config/browsers/remote_browsers.json [hunk 4]
```diff
@@ -1,34 +1,42 @@
 {
+  "bs_mac_safari": {
+    "browserName": "Safari",
+    "browser_version": "11.0",
+    "os": "OS X",
+    "os_version": "High Sierra",
+    "resolution": "1024x768",
```
Comment (LightGuard) [inline _tests/e2e/config/browsers/remote_browsers.json]: If we end up using this for visual checks, we should probably include some other resolutions as well, but that's probably down the road a bit.

Comment (RubyGemTSL) [reply _tests/e2e/config/browsers/remote_browsers.json]: Yeah don't think we need to worry about that at the moment. These are just browserstack capabilities. They can easily be reconfigured.

**Comment by RubyGemTSL** [comment]
retest this please

**Comment by RubyGemTSL** [comment]
@LightGuard I ended up having to do a bit more work on this after your review. it's been through and passed once, I'm just retesting to double check then it should be good to merge. Thank's.

**Comment by RubyGemTSL** [comment]
All good if you're happy @LightGuard

File: _docker/tests/test-runner/test_run_e2e_tests_options.rb [hunk 5]
```diff
@@ -113,12 +113,12 @@ def test_docker_execution_specifying_update_github_status
 
   def test_docker_execution_specifying_mobile_device
 
-    test_configuration = @run_tests_options.parse_command_line(%w(--e2e --base-url=http://foo.com --use-docker --browser=iphone_6))
+    test_configuration = @run_tests_options.parse_command_line(%w(--e2e --base-url=http://foo.com --use-docker --browser=iphone_8))
 
     assert(test_configuration[:docker])
-    assert_equal('iphone_6', test_configuration[:browser])
+    assert_equal('iphone_8', test_configuration[:browser])
```
Comment (LightGuard) [inline _docker/tests/test-runner/test_run_e2e_tests_options.rb]: Is the latest version of the phone (yes, I know the X is out) what we want to be testing against all the time? Seems a bit bleeding edge to me, but maybe there's a good reason

File: _tests/Dockerfile [hunk 6]
```diff
@@ -1,7 +1,7 @@
 FROM centos:7.3.1611
 MAINTAINER ihamilto@redhat.com
 
-RUN yum install -y  epel-release gcc gcc-c++ git libcurl-devel libffi-devel libtool libxml2 libxml2-devel libxslt \
+RUN yum clean all && yum install -y  epel-release gcc gcc-c++ git libcurl-devel libffi-devel libtool libxml2 libxml2-devel libxslt \
```
Comment (LightGuard) [inline _tests/Dockerfile]: :) Sorry, the clean should be the last command. The layered file system of Docker takes a snapshot of the filesystem at the end of each command and that becomes the new layer. You want those layers to be as small as possible when the command finishes. In this case, we'd want to clean up any yum caches, temp files, or downloaded files at the end of the run command so we only have on that layer what is absolutely necessary. I hope that makes more sense.

File: _tests/e2e/Dockerfile (hunk 3 above)
Comment (LightGuard) [reply _tests/e2e/Dockerfile]: Same thing here.

Comment (LightGuard) [reply _tests/e2e/Dockerfile]: We'd also want to remove the downloaded tarball at the end of this string of commands.

**Comment by RubyGemTSL** [comment]
@Dantheman720 that failing test is not related to your PR. I finally found the root cause of that failure, and was fixed as part of #2145.

**Pull request merged**